rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/RascalLanguageServer.java

```diff
@@ -98,17 +107,29 @@ private static void startLSP(Launcher<LanguageClient> server) {
         }
     }
 
-    private static final class ActualLanguageServer implements LanguageServer, LanguageClientAware {
+    public static final class ActualLanguageServer implements LanguageServer, LanguageClientAware, IRascalLanguageServerExtensions {
         private static final Logger logger = LogManager.getLogger(ActualLanguageServer.class);
         private final RascalTextDocumentService lspDocumentService;
         private final RascalWorkspaceService lspWorkspaceService = new RascalWorkspaceService();
         private final Runnable onExit;
+        private IRascalLanguageClient client;
+        private IDEServicesConfiguration ideServicesConfiguration;
 
         private ActualLanguageServer(Runnable onExit) {
             this.onExit = onExit;
             lspDocumentService = new RascalTextDocumentService(new RascalLanguageServices());
         }
 
+        @Override
+        public CompletableFuture<IDEServicesConfiguration> supplyIDEServicesConfiguration() {
+            if (ideServicesConfiguration != null) {
+                return CompletableFuture.completedFuture(ideServicesConfiguration);
+            }
+            
+            logger.log(Level.ERROR, "no IDEServicesConfiguration is set?");
```

Comment:
you can use: `logger.error`, also, if you throw an exception afterwards, you don't really need to print, that error will be caught somewhere and reported.

Reply:
check